src/librustc_trans/trans/_match.rs

```diff
@@ -1399,12 +1418,20 @@ fn trans_match_inner<'blk, 'tcx>(scope_cx: Block<'blk, 'tcx>,
     // `compile_submatch` works one column of arm patterns a time and
     // then peels that column off. So as we progress, it may become
     // impossible to tell whether we have a genuine default arm, i.e.
-    // `_ => foo` or not. Sometimes it is important to know that in order
-    // to decide whether moving on to the next condition or falling back
-    // to the default arm.
+    //      _ => foo
+    //      (_, _) => foo
+    // or not. Sometimes it is important to know that in order to decide
+    // whether moving on to the next arm or falling back to the default
+    // one.
+    fn all_wild_pat(pat: &ast::Pat) -> bool {
+        match pat.node {
+            ast::PatWild(ast::PatWildSingle) => true,
+            ast::PatTup(ref pats) => pats.iter().all(|p| all_wild_pat(&**p)),
+            _ => false,
+        }
+    }
```

Comment:
@jakub-, it works expect that a constant pattern such as:

```rust
  enum A { AA }
  const B: A = A::AA;
  match A::AA {
    B => ()
  }
```
triggers `const pattern should've been rewritten` error condition in your `check_match::pat_constructors` function. I'm not terribly familiar with it. Any thoughts?